src/Compilers/CSharp/Portable/Lowering/MethodToClassRewriter.cs

```diff
@@ -566,6 +573,15 @@ private PropertySymbol VisitPropertySymbol(PropertySymbol property)
                 return null;
             }
 
+            if (property.IsInExtensionClass)
+            {
+                // All incoming extension properties should be reduced
+                Debug.Assert(property.IsStatic);
```

Comment:
This code was very incorrect, and I have fixed it. The issue was that BoundAwaitExpression wasn't getting rewritten in LocalRewriter, which caused some really strange side effects that were difficult to understand (and I thought this spot was the right point to handle it).